packages/core/test/scuttle.spec.js

```diff
@@ -0,0 +1,126 @@
+    'To learn more visit https://github.com/LavaMoat/LavaMoat/pull/462.',
+}
+
+const err = (intrinsic) =>
+  'LavaMoat - property "' +
+  intrinsic +
+  '" of globalThis is inaccessible under ' +
+  'scuttling mode. To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.'
+
+function testScuttling(
+  globalRef,
+  intrinsics = ['Array', 'Int8Array', 'RegExp']
+) {
+  return intrinsics.some((intrinsic) => {
+    try {
+      const _ = globalRef[intrinsic]
+      return false
+    } catch (e) {
+      if (err(intrinsic) === e.message) {
+        return true
+      }
+      throw new Error('Unknown error thrown at testScuttling: ' + e.message)
+    }
+  })
+}
+
+test('scuttle - no opts', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis)
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+})
+
+test('scuttle - opts as bool', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, false)
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, true)
+  t.is(
+    testScuttling(vmGlobalThis),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+})
+
+test('scuttle - opts as object', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, { enabled: false })
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, { enabled: true })
+  t.is(
+    testScuttling(vmGlobalThis),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+})
+
+test('scuttle - exceptions', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  scuttle(vmGlobalThis, {
+    enabled: true,
+    exceptions: ['/[a-zA-Z0-8]*Array/', 'RegExp'],
+  })
+  t.is(
+    testScuttling(vmGlobalThis, ['String']),
+    true,
+    'expect intrinsics to not be accessible'
+  )
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+})
+
+test('scuttle - scuttle func', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  t.is(testScuttling(vmGlobalThis), false, 'expect intrinsics to be accessible')
+  let globalRef
+  const cb = (realm, scuttle) => ((globalRef = realm), scuttle(realm))
+  t.throws(
+    (_) => scuttle(vmGlobalThis, { enabled: true, scuttlerName: 'SCUTTLER' }),
+    SCUTTLER_NAME_ERROR
+  )
+  Object.defineProperty(vmGlobalThis, 'SCUTTLER', { value: cb })
```

Comment:
wait, so, the scuttler, in addition to a function passed to `scuttle`, can just be a function on the `GlobalRef`?

Reply:
Not exactly, opts do not respect non "stringifyable", because in the browser version, this gets bundled, so you can't really pass a function.
`scuttlerName` expects a string to represent a property expected to be found on the global object, which its value should be the `scuttlerFunc` function.
Not great, but this is a clear limitation we have to respect if we want LavaMoat to work for browser envs.
We can consider other ways to do this, but the limitation itself isn't something we can play around I think.

Reply:
IMO these guardrails need to be more clearly-defined via comments and/or types and/or exceptions. Whether that should be part of this PR is up to you

By "this gets bundled" you mean how the kernel uses templates to import source code as strings, or something else?  Trying to follow

Reply:
like this 0291d093c?